service/controller/app/v1/resource/configmap/desired.go

```diff
@@ -77,7 +77,12 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	return configMap, nil
 }
 
-func (r *Resource) getConfigMap(ctx context.Context, configMapName, configMapNamespace string) (*corev1.ConfigMap, error) {
+func (r *Resource) getConfigMap(ctx context.Context, configMapName, configMapNamespace string) (map[string]string, error) {
+	if configMapName == "" {
+		// Return early.
+		return map[string]string{}, nil
```

Comment:
Does it has to be an initialized empty map? Zero value would be nil if no other reason applies to do otherwise.

Reply:
Good point on the zero value. The data will be run through `helmclient.MergeValues` so the empty map helps there. But actually I think `MergeValues` should handle nil values.

I can make it nil and also update `helmclient`. WDYT?

Reply:
Code in general should be able to deal with zero values. Makes it more resilient and easier to comprehend here. Otherwise it looks like there is some deeper meaning to it that nobody gets.

Reply:
Good point, thanks. I'm going to change `helmclient.MergeValues` to handle nils and then simplify here by returning nil.

Reply:
Returning nil works without needing changes to helmclient. So going with that to simplify.